crates/cargo-util-schemas/src/core/partial_version.rs

```diff
@@ -163,6 +159,22 @@ impl<'de> serde::Deserialize<'de> for PartialVersion {
     }
 }
 
+/// Error parsing a [`PartialVersion`].
+#[derive(Debug, thiserror::Error)]
+pub enum PartialVersionError {
+    #[error("unexpected version requirement, expected a version like \"1.32\"")]
+    VersionReq,
+
+    #[error("unexpected prerelease field, expected a version like \"1.32\"")]
+    Prerelease,
+
+    #[error("unexpected build field, expected a version like \"1.32\"")]
+    BuildMetadata,
+
+    #[error("{0}, expected a version like \"1.32\"")]
+    SemVer(#[from] semver::Error),
```

Comment:
I'm not generally a fan of having error variants that wrap other errors because which and when those errors occur is dependent on the implementation

Reply:
This is removed. Other wrappers for errors inside this package are kept, as we have controls over those types.

Reply:
Now that the enum is private, I don't care as much